packages/headless/src/controllers/generated-answer/headless-generated-answer.ts

```diff
@@ -30,11 +31,17 @@ export function buildGeneratedAnswer(
   engine: SearchEngine,
   props: GeneratedAnswerProps = {}
 ): GeneratedAnswer {
-  const controller = buildSearchAPIGeneratedAnswer(
-    engine,
-    generatedAnswerAnalyticsClient,
-    props
-  );
+  const controller = props.answerConfigurationId
+    ? buildKnowledgeGeneratedAnswer(
+        engine,
+        generatedAnswerAnalyticsClient,
+        props
+      )
+    : buildSearchAPIGeneratedAnswer(
+        engine as SearchEngine,
```

Comment:
why the `as SearchEngine` is needed here? the engine is already of `SearchEngine` in this controller,

Reply:
actually we need it because the `SearchAPIGeneratedAnswer` controller will be used by the insight engine. We pass the insight engine to the `buildGeneratedAnswer` but I had to cast the `searchEngine` when passing it to the `buildSearchAPIGeneratedAnswer`